src/CMakeLists.txt

```diff
@@ -91,65 +91,65 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
     endif()
 endif()
 
-### Installation (https://github.com/forexample/package-example) {
-
-# Introduce variables:
-# * CMAKE_INSTALL_LIBDIR
-# * CMAKE_INSTALL_BINDIR
-# * CMAKE_INSTALL_INCLUDEDIR
-include(GNUInstallDirs)
+add_library(${TARGET_NAME} ${MKLDNN_LIBRARY_TYPE} ${HEADERS} ${SOURCES})
+set(TARGET_EXPORT_NAME "${TARGET_NAME}-targets")
+string(TOUPPER "${TARGET_NAME}::" TARGET_NAMESPACE_NAME)
+set_property(TARGET ${TARGET_NAME} PROPERTY CXX_STANDARD 11)
+set_property(TARGET ${TARGET_NAME} PROPERTY CXX_STANDARD_REQUIRED ON)
+set_property(TARGET ${TARGET_NAME} PROPERTY VERSION "${PROJECT_VERSION}.0")
+set_property(TARGET ${TARGET_NAME} PROPERTY SOVERSION "0")
+set_property(TARGET ${TARGET_NAME} PROPERTY PUBLIC_HEADER ${HEADERS})
 
-set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${LIB_NAME}")
-set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
-set(version_config "${generated_dir}/${LIB_NAME}ConfigVersion.cmake")
-set(project_config "${generated_dir}/${LIB_NAME}Config.cmake")
-set(TARGETS_EXPORT_NAME "${LIB_NAME}Targets")
-set(namespace "${LIB_NAME}::")
+target_include_directories(${TARGET_NAME} PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/../include>
+    # $<INSTALL_PREFIX> is required for compatibility with cmake 2.8
+    $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}>
+    )
 
-include(CMakePackageConfigHelpers)
+# Link to static and shared libs with full paths privately during build time
+# This is rather ugly, but required for the generator expressions to
+# expand correctly
+foreach(l IN LISTS EXTRA_SHARED_LIBS EXTRA_STATIC_LIBS)
+    target_link_libraries(${TARGET_NAME} LINK_PRIVATE $<BUILD_INTERFACE:${l}>)
+endforeach()
```

Comment:
Actually: Why not combine both loops and strip in the loop? This makes it much more clear what is being done here and is shorter

Reply:
The loops iterate over different lists, so I'm not sure I understand what you mean here.

Reply:
Ah yes, you have `EXTRA_STATIC_LIBS` in the first loop. The idea was to get rid of the `strip_paths` call and simply do the extraction in 1 loop which simplifies reasoning on what happens. With that extra list you'd still need 2 loops though. Might still be better but well...

Reply:
Your solution looks good too (without the loops)